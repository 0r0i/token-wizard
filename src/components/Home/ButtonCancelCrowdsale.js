import React, { Component } from 'react'
import logdown from 'logdown'
import cancelDeploy from '../../utils/cancelDeploy'

const logger = logdown('TW:Home:CancelCrowdsale')

export class ButtonCancelCrowdsale extends Component {
  cancel = () => {
    logger.log('Cancel crowdsale')
    cancelDeploy()
  }

  render() {
    return (
      <button onClick={() => this.cancel()} className="hm-ButtonCancelCrowdsale">
        Cancel crowdsale
      </button>
    )
  }
}
