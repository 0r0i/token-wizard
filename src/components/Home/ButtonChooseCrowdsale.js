import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'

export class ButtonChooseCrowdsale extends Component {
  goToCrowdsales = () => {
    localStorage.reload = true
    navigateTo({
      history: this.props.history,
      location: 'crowdsales',
      fromLocation: 'home'
    })
  }

  render() {
    return (
      <div onClick={() => this.goToCrowdsales()} className="hm-ButtonChooseCrowdsale">
        Choose Crowdsale
      </div>
    )
  }
}
