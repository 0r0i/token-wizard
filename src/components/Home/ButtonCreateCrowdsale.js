import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'

export class ButtonCreateCrowdsale extends Component {
  goToStepOne = () => {
    localStorage.reload = true
    navigateTo({
      history: this.props.history,
      location: 'stepOne',
      fromLocation: 'home'
    })
  }

  render() {
    return (
      <button onClick={() => this.goToStepOne()} className="hm-ButtonCreateCrowdsale">
        New crowdsale
      </button>
    )
  }
}
