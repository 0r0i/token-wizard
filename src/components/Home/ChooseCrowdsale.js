import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'

export class ChooseCrowdsale extends Component {
  goToCrowdsales = () => {
    navigateTo(this.props, 'crowdsales')
  }

  render() {
    return (
      <div onClick={() => this.goToCrowdsales()} className="hm-Home_BtnChoose">
        Choose Crowdsale
      </div>
    )
  }
}
