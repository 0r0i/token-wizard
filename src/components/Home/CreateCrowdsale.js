import React, { Component } from 'react'
import { navigateTo } from '../../utils/utils'
import logdown from 'logdown'

const logger = logdown('TW:Home:CreateCrowdsale')

export class CreateCrowdsale extends Component {
  goToStepOne = () => {
    try {
      localStorage.reload = true
      navigateTo({
        history: this.props.history,
        location: 'stepOne',
        fromLocation: 'home'
      })
    } catch (err) {
      logger.log('Error to navigate', err)
    }
  }

  render() {
    return (
      <button onClick={() => this.goToStepOne()} className="hm-Home_BtnNew">
        New crowdsale
      </button>
    )
  }
}
