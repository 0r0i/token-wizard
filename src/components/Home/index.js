import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import CrowdsalesList from '../Common/CrowdsalesList'
import { Loader } from '../Common/Loader'
import { loadRegistryAddresses } from '../../utils/blockchainHelpers'
import { ModalContainer } from '../Common/ModalContainer'
import { toast } from '../../utils/utils'
import { TOAST, NAVIGATION_STEPS, DOWNLOAD_STATUS } from '../../utils/constants'
import { inject, observer } from 'mobx-react'
import { checkWeb3, getNetworkVersion } from '../../utils/blockchainHelpers'
import { getCrowdsaleAssets } from '../../stores/utils'
import logdown from 'logdown'
import { LogoPrimary } from '../LogoPrimary/index'

const logger = logdown('TW:home')

const { CROWDSALE_STRATEGY, TOKEN_SETUP, CROWDSALE_SETUP, PUBLISH, CROWDSALE_PAGE } = NAVIGATION_STEPS

@inject('web3Store', 'generalStore', 'contractStore')
@observer
export class Home extends Component {
  constructor(props) {
    super(props)
    this.state = {
      showModal: false,
      loading: false
    }

    let { contractStore } = this.props
    contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.PENDING)
  }

  componentDidMount() {
    let { generalStore, web3Store, contractStore } = this.props
    checkWeb3(web3Store.web3)

    getNetworkVersion()
      .then(networkID => {
        generalStore.setProperty('networkID', networkID)
        getCrowdsaleAssets(networkID)
      })
      .then(
        () => {
          contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.SUCCESS)
        },
        e => {
          logger.error('Error downloading contracts', e)
          toast.showToaster({
            type: TOAST.TYPE.ERROR,
            message:
              'The contracts could not be downloaded.Please try to refresh the page. If the problem persists, try again later.'
          })

          contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.FAILURE)
        }
      )
  }

  chooseContract = () => {
    this.setState({
      loading: true
    })

    loadRegistryAddresses().then(
      () => {
        this.setState({
          loading: false,
          showModal: true
        })
      },
      e => {
        logger.error('There was a problem loading the crowdsale addresses from the registry', e)
        this.setState({
          loading: false
        })
      }
    )
  }

  onClick = crowdsaleAddress => {
    this.props.history.push('/manage/' + crowdsaleAddress)
  }

  hideModal = () => {
    this.setState({ showModal: false })
  }

  render() {
    return (
      <div>
        <section className="hm-Home">
          <div className="hm-Home_Scroll">
            <div className="hm-Home_Top">
              <LogoPrimary />
            </div>
            <div className="hm-Home_MainInfoContainer">
              <div className="hm-Home_MainInfo">
                <div className="hm-Home_MainInfoTextContainer">
                  <h1 className="hm-Home_MainInfoTitle">Welcome to Token Wizard</h1>
                  <p className="hm-Home_MainInfoDescription">
                    Token Wizard is a client side tool to create ERC20 token and crowdsale in five steps. It helps you
                    to publish contracts on the Ethereum network, create a crowdsale page with stats. For participants,
                    the wizard creates a page to contribute into the campaign.
                  </p>
                  <p className="hm-Home_MainInfoPoweredBy">
                    Token Wizard is powered by <a href="https://github.com/auth-os/beta">Auth-os</a>.
                  </p>
                </div>
                <div className="hm-Home_MainInfoButtonContainer">
                  <Link className="hm-Home_BtnNew" to="/1">
                    New crowdsale
                  </Link>
                  <div onClick={() => this.chooseContract()} className="hm-Home_BtnChoose">
                    Choose Crowdsale
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        {/* <ModalContainer
          title={'Crowdsale List'}
          description={`The list of your updatable crowdsales. Choose crowdsale address, click Continue and you'll be
        able to update the parameters of crowdsale.`}
          // hideModal={this.hideModal}
          // showModal={this.state.showModal}
          showModal={true}
        >
          <CrowdsalesList onClick={this.onClick} />
        </ModalContainer> */}
        <Loader show={this.state.loading} />
      </div>
    )
  }
}
