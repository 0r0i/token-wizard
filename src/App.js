import React, { Component } from 'react'
import { inject, observer } from 'mobx-react'
import { Home, Manage, stepOne, stepTwo, stepThree, stepFour, Crowdsale, Contribute, Stats } from './components/index'
import NoWeb3 from './components/Common/NoWeb3'
import IncompleteDeploy from './components/IncompleteDeploy'
import { getQueryVariable } from './utils/utils'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import AlertContainer from 'react-alert'
import { TOAST } from './utils/constants'
import { toast } from './utils/utils'
import { Web3Provider } from './react-web3'

// import './assets/stylesheets/application.css'

@inject('deploymentStore')
@observer
class App extends Component {
  componentWillMount() {
    // temporary code to switch ui styling versions
    // obviously delete this when we get the v2.0 up and running
    const uiversion = getQueryVariable('uiversion')

    if (uiversion === '2') {
      require('./assets/stylesheets/styles.css')
    } else if (uiversion === '1') {
      require('./assets/stylesheets/application_styles.css')
    } else {
      require('./assets/stylesheets/application_styles.css')
      require('./assets/stylesheets/styles.css')
    }
  }
  render() {
    const { deploymentStore } = this.props
    var crowdsaleAddr = getQueryVariable('addr')

    return (
      <Router>
        <div>
          {/* <Header /> */}

          <Switch>
            <Route exact path="/crowdsale" component={Crowdsale} />
            <Route exact path="/contribute" component={Contribute} />
            <Route exact path="/stats" component={Stats} />
            <Route>
              <Web3Provider onChangeAccount={deploymentStore.handleAccountChange} web3UnavailableScreen={NoWeb3}>
                <Switch>
                  {/* The route to /4 must be first for the incomplete deploy redirect to work */}
                  <Route path="/4" component={stepFour} />

                  {deploymentStore.deployInProgress ? (
                    <IncompleteDeploy />
                  ) : (
                    <Switch>
                      <Route exact path="/" component={crowdsaleAddr ? Crowdsale : Home} />
                      <Route exact path="/manage/:crowdsalePointer" component={Manage} />
                      <Route path="/1" component={stepOne} />
                      <Route path="/2" component={stepTwo} />
                      <Route path="/3" component={stepThree} />
                    </Switch>
                  )}
                </Switch>
              </Web3Provider>
            </Route>
          </Switch>

          {/* <Footer /> */}
          <AlertContainer ref={a => (toast.msg = a)} {...TOAST.DEFAULT_OPTIONS} />
        </div>
      </Router>
    )
  }
}

export default App
