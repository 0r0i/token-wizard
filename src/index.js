import React from 'react';
import ReactDOM from 'react-dom';
import Web3 from 'web3'
import App from './App';
import registerServiceWorker from './registerServiceWorker';
import { useStrict } from 'mobx';
import { Provider } from 'mobx-react';
import { REACT_PREFIX } from './utils/constants'
import * as stores from './stores';
import 'airbnb-js-shims'
import 'font-awesome/css/font-awesome.css'

useStrict(true);

//todo: remove unused contracts
checkEnvVariable(`${REACT_PREFIX}REGISTRY_IDX_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}PROVIDER_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}REGISTRY_EXEC_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}MINTED_CAPPED_IDX_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}MINTED_CAPPED_CROWDSALE_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}MINTED_CAPPED_CROWDSALE_MANAGER_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}MINTED_CAPPED_TOKEN_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}MINTED_CAPPED_TOKEN_MANAGER_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}DUTCH_IDX_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}DUTCH_CROWDSALE_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}DUTCH_CROWDSALE_MANAGER_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}DUTCH_TOKEN_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}DUTCH_TOKEN_MANAGER_ADDRESS`);
checkEnvVariable(`${REACT_PREFIX}MINTED_CAPPED_APP_NAME`);
checkEnvVariable(`${REACT_PREFIX}DUTCH_APP_NAME`);

const devEnvironment = process.env.NODE_ENV === 'development';
if (devEnvironment && !window.web3) {
  window.web3 = new Web3(new Web3.providers.HttpProvider('http://localhost:8545'));
}

function checkEnvVariable(envVar) {
  if (!process.env[envVar]) {
    throw new Error(`${envVar} env variable is not present`)
  }
}

function renderApp(App) {
  ReactDOM.render(
    <Provider { ...stores }>
      <App />
    </Provider>,
    document.getElementById('root')
  );
}

renderApp(App)

if (module.hot) {
  module.hot.accept(() => {
    const NextApp = require('./App').default
    renderApp(NextApp)
  })
}


registerServiceWorker();
