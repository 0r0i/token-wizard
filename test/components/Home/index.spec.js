import React from 'react'
import Adapter from 'enzyme-adapter-react-15'
import { configure, mount, shallow, render } from 'enzyme'
import { MemoryRouter } from 'react-router'
import { Provider } from 'mobx-react'
import { Home } from '../../../src/components/Home/index'
import { ChooseCrowdsale } from '../../../src/components/Home/ChooseCrowdsale'
import { CreateCrowdsale } from '../../../src/components/Home/CreateCrowdsale'
import storage from 'store2'
import { deploymentStore } from '../../../src/stores'

configure({ adapter: new Adapter() })

describe('Home', () => {
  const history = {
    push: jest.fn()
  }
  const stores = { deploymentStore }

  it('should render screen with shallow without throwing an error', () => {
    // Given
    const wrapper = shallow(
      <Provider {...stores}>
        <Home />
      </Provider>
    )

    // When
    const tree = wrapper.find('div.hm-Home_MainInfo')

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('should render screen with mount without throwing an error', () => {
    // Given
    const wrapper = mount(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/']}>
          <Home />
        </MemoryRouter>
      </Provider>
    )

    // Given
    const tree = wrapper.html()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('should render screen with render without throwing an error', () => {
    // When
    const wrapper = render(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/']}>
          <Home />
        </MemoryRouter>
      </Provider>
    )

    // Given
    const tree = wrapper.html()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it(`should navigate to StepOne`, async () => {
    // Given
    const wrapper = mount(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/']}>
          <Home history={history} />
        </MemoryRouter>
      </Provider>
    )
    const createCrowdsaleComponentWrapper = wrapper.find(CreateCrowdsale)
    const instance = createCrowdsaleComponentWrapper.instance()
    const goToStepOneHandler = jest.spyOn(instance, 'goToStepOne')
    instance.forceUpdate()
    storage.clearAll()

    // When
    const buttonCreateCrowdsale = createCrowdsaleComponentWrapper.find('.hm-Home_BtnNew')
    buttonCreateCrowdsale.simulate('click')

    // Then
    expect(buttonCreateCrowdsale.length).toBe(1)
    expect(goToStepOneHandler).toHaveBeenCalledTimes(1)
    expect(goToStepOneHandler).toHaveBeenCalledWith()
  })

  it(`should navigate to Crowdsales List`, async () => {
    // Given
    const wrapper = mount(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/']}>
          <Home history={history} />
        </MemoryRouter>
      </Provider>
    )
    const chooseCrowdsaleComponentWrapper = wrapper.find(ChooseCrowdsale)
    const instance = chooseCrowdsaleComponentWrapper.instance()
    const goToCrowdsalesHandler = jest.spyOn(instance, 'goToCrowdsales')
    instance.forceUpdate()
    storage.clearAll()

    // When
    const chooseCrowdsale = chooseCrowdsaleComponentWrapper.find('.hm-Home_BtnChoose')
    chooseCrowdsale.simulate('click')

    // Then
    expect(chooseCrowdsale.length).toBe(1)
    expect(goToCrowdsalesHandler).toHaveBeenCalledTimes(1)
    expect(goToCrowdsalesHandler).toHaveBeenCalledWith()
  })
})
