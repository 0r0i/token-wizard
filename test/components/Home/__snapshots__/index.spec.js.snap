// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render Home screen 1`] = `
<div>
  <section
    className="hm-Home "
  >
    <div
      className="hm-Home_Scroll"
    >
      <div
        className="hm-Home_Top"
      >
        <a
          className="sw-LogoPrimary"
          href="/"
          onClick={[Function]}
        />
      </div>
      <div
        className="hm-Home_MainInfoContainer"
      >
        <div
          className="hm-Home_MainInfo"
        >
          <div
            className="hm-Home_MainInfoTextContainer"
          >
            <h1
              className="hm-Home_MainInfoTitle"
            >
              Welcome to Token Wizard
            </h1>
            <p
              className="hm-Home_MainInfoDescription"
            >
              Token Wizard is a client side tool to create ERC20 token and crowdsale in five steps. It helps you to publish contracts on the Ethereum network, create a crowdsale page with stats. For participants, the wizard creates a page to contribute into the campaign.
            </p>
            <p
              className="hm-Home_MainInfoPoweredBy"
            >
              Token Wizard is powered by 
              <a
                href="https://github.com/auth-os/beta"
              >
                Auth-os
              </a>
              .
            </p>
          </div>
          <div
            className="hm-Home_MainInfoButtonContainer"
          >
            <button
              className="hm-Home_BtnNew"
              onClick={[Function]}
            >
              New crowdsale
            </button>
            <div
              className="hm-Home_BtnChoose"
              onClick={[Function]}
            >
              Choose Crowdsale
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
